run/SU2_preCICE_CHT.py

```diff
@@ -170,19 +165,21 @@ def main():
   nTimeIter = SU2Driver.GetnTimeIter()
   time = TimeIter*deltaT
 
-  # Setup preCICE dt:
-  precice_deltaT = interface.initialize()
-
   # Set up initial data for preCICE
-  if (interface.is_action_required(precice.action_write_initial_data())):
+  if (participant.requires_initial_data()):
 
     for i, iVertex in enumerate(iVertices_CHTMarker_PHYS):
       read_data[i] = GetInitialFxn(CHTMarkerID, iVertex)
 
-    interface.write_block_scalar_data(write_data_id, vertex_ids, read_data)
-    interface.mark_action_fulfilled(precice.action_write_initial_data())
+    participant.write_data(mesh_name, precice_write, vertex_ids, read_data)
+
+  # Initialize preCICE
+  participant.initialize()
 
-  interface.initialize_data()
+  # Setup time step sizes:
+  precice_deltaT = participant.get_max_time_step_size()
+  deltaT = SU2Driver.GetUnsteady_TimeStep()
+  deltaT = min(precice_deltaT, deltaT)
 
   # Sleep briefly to allow for data initialization to be processed
   sleep(3)
```

Comment:
@j-signorelli is this actually needed, or was it there only for debugging purposes? For me, it seems to works without.

Reply:
It seems redundant calculation actual timestep calculated https://github.com/precice/su2-adapter/blob/980c07b616b9c69f3c7e8aad223328deaa0782df/run/SU2_preCICE_CHT.py#L215-L218 here and used for advancing SU2 Driver.

The same happens with the FSI adapter.

Reply:
I was talking about the `sleep(3)`, actually.

I think we still need to initialize `deltaT` for the first time step, because it is used later on.

Reply:
I see it is also used to retrieve data. I have always thought of Fluid as the first participant.

`sleep(3)` is required for SU2 on some systems, but it takes a while to partition mesh and allocate. Depending on your system, it might take unnecessarily long.

Reply:
@MakisH I did need to add the sleep statement. I remember back when writing this code and attempting to use precice data initialization, it was not working. Adding `sleep(3)` resolved the issue.

Reply:
Strange, it should not be the case. The initialization is blocking. If there are such issues, we should resolve them in preCICE, or understand why they are happening here.

I will remove the `sleep` and keep it in mind if anyone reports any issues. Thanks for the hints!

Reply:
Commented-out and added a comment to open an issue if this is needed.

Reply:
After reading all comments (I missed the one by @kursatyurt), I re-enabled by default, stating that it is only needed on some systems and use cases.

Reply:
@MakisH @kursatyurt I haven't personally ever seen that sleep statement used, and none of the [SU2 pywrapper test cases](https://github.com/su2code/SU2/tree/master/TestCases/py_wrapper) have them.

It may be useful to include a link or source to where this was heard or seen in the comment to avoid any confusion about that line by other users.